storage/bootstrap/result/types.go

```diff
@@ -46,6 +47,38 @@ type BootstrapResult interface {
 	SetUnfulfilled(unfulfilled ShardTimeRanges)
 }
 
+// IndexBootstrapResult is the result of a bootstrap of series index metadata.
+type IndexBootstrapResult interface {
+	// Blocks returns a map of all index block results.
+	IndexResults() IndexResults
+
+	// Unfulfilled is the unfulfilled time ranges for the bootstrap.
+	Unfulfilled() xtime.Ranges
+
+	// SetUnfulfilled sets the current unfulfilled shard time ranges.
+	SetUnfulfilled(unfulfilled xtime.Ranges)
+
+	// Add adds an index block result.
+	Add(block IndexBlock, unfulfilled xtime.Ranges)
+}
+
+// IndexResults is a set of index blocks indexed by block start.
+type IndexResults map[xtime.UnixNano]IndexBlock
+
+// IndexBlock contains the bootstrap data structures for an index block.
+type IndexBlock struct {
+	blockStart time.Time
+	segments   []segment.Segment
+	notIndexed map[string]SeriesNotIndexed // TODO(r): create map specifically for this case
```

Comment:
ended up creating this map for https://github.com/m3db/m3db/pull/578, can probably share it

Reply:
Can do, I'll open up a PR once that lands.